cmd/qgb/keys/p2p/p2p.go

```diff
@@ -0,0 +1,283 @@
+			if err != nil {
+				return err
+			}
+
+			logger := tmlog.NewTMLogger(os.Stdout)
+
+			initOptions := store.InitOptions{NeedP2PKeyStore: true}
+			isInit := store.IsInit(logger, config.home, initOptions)
+
+			// initialize the store if not initialized
+			if !isInit {
+				err := store.Init(logger, config.home, initOptions)
+				if err != nil {
+					return err
+				}
+			}
+
+			// open store
+			openOptions := store.OpenOptions{HasP2PKeyStore: true}
+			s, err := store.OpenStore(logger, config.home, openOptions)
+			if err != nil {
+				return err
+			}
+			defer func(s *store.Store, log tmlog.Logger) {
+				err := s.Close(log, openOptions)
+				if err != nil {
+					logger.Error(err.Error())
+				}
+			}(s, logger)
+
+			var nickname string
+			if len(args) == 0 {
+				k, err := s.P2PKeyStore.List()
+				if err != nil {
+					return err
+				}
+				nickname = string(rune(len(k)))
+			} else {
+				nickname = args[0]
+			}
+
+			fmt.Println("generating a new Ed25519 private key", "nickname", nickname)
+
+			sr := util.NewTimeSeededRand()
+			priv, _, err := crypto.GenerateEd25519Key(sr)
+			if err != nil {
+				fmt.Println(err.Error())
+				return err
+			}
+
+			err = s.P2PKeyStore.Put(nickname, priv)
+			if err != nil {
+				return err
+			}
+
+			logger.Info("key created successfully", "nickname", nickname)
+			return nil
+		},
+	}
+	return keysConfigFlags(&cmd)
+}
+
+func List() *cobra.Command {
+	cmd := cobra.Command{
+		Use:   "list",
+		Short: "list existing p2p addresses",
+		Args:  cobra.ExactArgs(0),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			grandParentName := cmd.Parent().Parent().Parent().Use
+			serviceName, err := common.CommandToServiceName(grandParentName)
+			if err != nil {
+				return err
+			}
+			config, err := parseKeysConfigFlags(cmd, serviceName)
+			if err != nil {
+				return err
+			}
+
+			logger := tmlog.NewTMLogger(os.Stdout)
+
+			initOptions := store.InitOptions{NeedP2PKeyStore: true}
+			isInit := store.IsInit(logger, config.home, initOptions)
+
+			// check if not initialized
+			if !isInit {
+				logger.Info("p2p store not initialized", "path", config.home)
+				return store.ErrNotInited
+			}
+
+			// open store
+			openOptions := store.OpenOptions{HasP2PKeyStore: true}
+			s, err := store.OpenStore(logger, config.home, openOptions)
+			if err != nil {
+				return err
+			}
+			defer func(s *store.Store, log tmlog.Logger) {
+				err := s.Close(log, openOptions)
+				if err != nil {
+					logger.Error(err.Error())
+				}
+			}(s, logger)
+
+			logger.Info("the p2p keys nicknames list:")
+
+			l, err := s.P2PKeyStore.List()
+			if err != nil {
+				return err
+			}
+
+			for _, k := range l {
+				logger.Info(k)
+			}
+
+			return nil
+		},
+	}
+	return keysConfigFlags(&cmd)
+}
+
+func Import() *cobra.Command {
+	cmd := cobra.Command{
+		Use:   "import <nickname> <private_key_in_hex_without_0x>",
+		Short: "import an existing p2p private key",
```

Comment:
perhaps a stupid question, but when would we ever want to import p2p keys instead of regenerate them? Do we currently have the ability to selectively connect to specific peers? 

I've done this before for sentry node setups for a validator, but is that type of setup something we plan on supporting?